### ISSUE TITLE
[WebSocket Transport]: aggregate or cork writes to underlying transport (IDFGH-14105)

### ISSUE THREAD

**Issue by bryghtlabs-richard** [espressif/esp-idf#14914]
### Is your feature request related to a problem?

WebSocket writes (eventually calling tranport_ws.c's _ws_write() function) tend to create more packets than needed. This increases single packet upload latency from 1 packet to 3 packets:

1. Client->Server: TLS send WebSocket header, Nagler blocks payload upload
2. Server->Client: TCP ACK for WebSocket header packet, unblock Nagler
3. Client->Server: TLS send WebSocket payload

### Describe the solution you'd like.

I'd like _ws_write() to either batch writes to `esp_transport_write()`, or to add something like `esp_transport_cork()` and `esp_transport_push()` to the transport layer so that higher layer transports can combine writes that would then be sent  only when `esp_transport_push()` is called.

### Describe alternatives you've considered.

Nagle's algorithm doesn't help here - the first TLS packet containing the WebSocket frame headers is sent immediately, and the TLS packet containing the WebSocket payload deferred until the other side aknowledges the first TLS packet. Once the acknowledgement is received, the Nagler is disabled and the payload is uploaded.

### Additional context.

![image](https://github.com/user-attachments/assets/f7030dc9-7d28-4d33-a0b5-2b2cd92809ee)

**Comment by bryghtlabs-richard** [comment]
I've also requested https://github.com/Mbed-TLS/mbedtls/issues/9846 from MbedTLS, which will enable optimizing this without buffering before the TLS layer.
